### PR TITLE
[docs] Add new Python SchemaTransform multi-lang guide

### PR DESCRIPTION
**Pull request by ahmedabu98**
Part of #33358

Adding a new page to the Beam site

### CODE REVIEW

File: website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    annotationProcessor "com.google.auto.service:auto-service:1.0.1"
+    // AutoValue annotation for our configuration object
+    annotationProcessor "com.google.auto.value:auto-value:1.9"
+}
+```
+
+Next, run the shaded JAR file, and provide a port to host the service. A list of available SchemaTransformProviders will be displayed.
+
+```shell
+$ java -jar path/to/my-expansion-service.jar 12345
```
Comment (liferoad) [inline website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md:190]: only this is shell, right?

File: website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+inspect.getdoc(MyTransform)
+# Output: "This transform does this and that..."
+
+inspect.signature(MyTransform)
+# Output: (foo: "str: Description of what foo does...",
+#	     bar: "int: Description of what bar does....")
+```
+
+This metadata is generated directly from the provider's implementation. The class documentation is generated from the [optional **description** method](#additional-metadata). The signature information is generated from the `@SchemaFieldDescription` annotations in the [configuration object](#implement-a-configuration).
+
```
Comment (liferoad) [inline website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md:272]: do we have a full example somewhere for this guide?

File: website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md (hunk 1 above)
Comment (ahmedabu98) [reply website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md:190]: Yes this is in the command line

File: website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md (hunk 2 above)
Comment (ahmedabu98) [reply website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md:272]: I'm adding a quickstart that uses an example with the same format here #33360

Comment (liferoad) [reply website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md:272]: make sure we link it here.

File: website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+To streamline this process, Beam’s portability framework enables the use of portable transforms that can be shared across languages. This reduces duplication, allowing developers to focus on maintaining only N transforms. Pipelines combining [portable transforms](#portable-transform) from other SDK(s) are known as [“multi-language” pipelines](../programming-guide.md#13-multi-language-pipelines-multi-language-pipelines).
+
+The SchemaTransform framework represents the latest advancement in enhancing this multi-language capability.
+
+The following jumps straight into the guide. Check out the [appendix](#appendix) section below for some of the terminology used here. For a runnable example, check out this [page](python-multi-language-pipelines-2.md).
```
Comment (ahmedabu98) [inline website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md:29]: FYI @liferoad the quickstart is linked here

**Pull request merged**